src/components/panel/panel.spec.js

```diff
@@ -8,37 +9,105 @@ describe('$mdPanel', function() {
     $mdPanel = $injector.get('$mdPanel');
     $rootScope = $injector.get('$rootScope');
+    $rootEl = $injector.get('$rootElement');
   };
 
   beforeEach(function() {
     module('material.components.panel', 'ngSanitize');
 
     inject(injectLocals);
+
+    // By default, the panel is attached to $rootElement, so add it to the DOM.
+    attachToBody($rootEl);
+  });
+
+  afterEach(function() {
+    attachedElements.forEach(function(el) {
+      el.remove();
+    });
+    attachedElements = [];
   });
 
-  it('creates a basic panel', function() {
-    var resolved = false;
-    var config = {attachTo: 'hello'};
+  it('should create a basic panel', function() {
+    var openResolved = false;
+    var closeResolved = false;
+    var config = {};
 
     var panelRef = $mdPanel.create(config);
 
     expect(panelRef.id).toBeDefined();
-    expect(panelRef.open).toBeDefined();
-    expect(panelRef.close).toBeDefined();
-    expect(panelRef.config).toEqual(config);
+    expect(panelRef.open).toBeOfType('function');
+    expect(panelRef.close).toBeOfType('function');
     expect(panelRef.isOpen).toEqual(false);
 
     panelRef.open().then(function() {
-      resolved = true;
+      openResolved = true;
     });
     $rootScope.$apply();
 
+    expect(openResolved).toBe(true);
     expect(panelRef.isOpen).toEqual(true);
 
-    panelRef.close();
+    panelRef.close().then(function() {
+      closeResolved = true;
+    });
     $rootScope.$apply();
 
-    expect(resolved).toBe(true);
+    expect(closeResolved).toBe(true);
     expect(panelRef.isOpen).toEqual(false);
   });
+
+  it('should add and remove a panel from the DOM', function() {
+    var template = '<div id="panel">Hello World!</div>';
+    var config = { template: template };
+
+    var panelRef = $mdPanel.create(config);
+
+    expect('#panel').not.toExist();
+
+    panelRef.open();
+    $rootScope.$apply();
+
+    expect('#panel').toExist();
+
+    panelRef.close();
+    $rootScope.$apply();
+
+    expect('#panel').not.toExist();
+  });
+
+  it('should attach panel to a specific element', function() {
+    var parentEl = document.createElement('div');
+    parentEl.id = 'parent';
+    attachToBody(parentEl);
+
+    var template = '<div id="panel">Hello World!</div>';
+    var config = {
+      attachTo: angular.element(parentEl),
+      template: template
+    };
+
+    var panelRef = $mdPanel.create(config);
+    panelRef.open();
+    $rootScope.$apply();
+
+    var panelEl = document.querySelector('#panel');
+    expect(panelEl.parentElement).toBe(parentEl);
+
+    panelRef.close();
+    $rootScope.$apply();
+
+    expect(parentEl.childElementCount).toEqual(0);
+  });
+
+  /**
+   * Attached an element to document.body. Keeps track of attached elements
+   * so that they can be removed in an afterEach.
+   * @param el
+   */
+  function attachToBody(el) {
+    var element = angular.element(el);
+    angular.element(document.body).append(element);
```

Comment:
You don't have to change anything here, but in general I'm in favor of using native DOM APIs instead of angular's JQLite APIs whenever possible. The JQLite stuff is a remnant of an era when even things like `addEventListener` weren't totally standard; these days the native APIs will be what you need 90% of the time.